addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -90,6 +92,11 @@ def _process_step_create_lead_and_forward(self, discuss_channel):
                 lambda team: team.assignment_max
                 and lead.filtered_domain(literal_eval(team.assignment_domain or "[]"))
             )
+        if self.env.user.partner_id.company_id:
```

Comment:
Can't you use ``_get_default_team_id`` defined in 'sales_team' module (here and/or above) ? a bit same reasoning, we try to avoid as much "custom search" as possible when searching for team.

Reply:
It does not look like the right method for this job.

The first part of this method checks for the current user which is not what we want in this case (the current user is the visitor here, not the sales person).

And the user param of the method is the target sales person but at this step in the flow we have not assigned a sales person yet, we're only trying to find which teams are "possible", and then we select an available person from those teams based on availability for live chat.

On top of that, we want to follow assignment_domain/assignment_optout which the "default" method ignores.

Reply:
Actually it was used for company filtering, not user, read a bit fast :) .